cli/inagoctl.go

```diff
@@ -34,19 +38,34 @@ var (
 			// This callback is executed after flags are parsed and before any
 			// command runs.
 
+			logLevel := "INFO"
+			if globalFlags.Verbose {
+				logLevel = "DEBUG"
+			}
+
+			logger := requestcontext.MustGetLogger(
+				requestcontext.LoggerConfig{
+					Name:                "inago",
+					Level:               logLevel,
+					Color:               isatty.IsTerminal(os.Stderr.Fd()),
+					IncludeNameInFormat: false,
+				},
+			)
+			newLogger = &logger
+
 			URL, err := url.Parse(globalFlags.FleetEndpoint)
 			if err != nil {
 				panic(err)
 			}
 
-			newFleetConfig := fleet.DefaultConfig()
+			newFleetConfig := fleet.DefaultConfig(newLogger)
```

Comment:
-1 

The default config is there to provide some default configuration based on best effort of the packages knowledge. This should not receive any argument. Instead the config returned by `DefaultConfig` should be changed. See the line below where we modify the `Endpoint` property.